cmake/qemu/CMakeLists.txt

```diff
@@ -57,8 +53,11 @@ if(EMU_PLATFORM)
           -pidfile qemu-${target}.pid
           )
       else()
+        # QEMU_INSTANCE is a command line argument to *make* (not cmake). By
+        # appending the instance name to the pid file we can easily run more
+        # instances of the same sample.
         list(APPEND MORE_FLAGS_FOR_${target}
-          -serial unix:/tmp/slip.sock${QEMU_INSTANCE}
+          -serial unix:/tmp/slip.sock\${QEMU_INSTANCE}
```

Comment:
Ah, so adding this '\\' character will make QEMU_INSTANCE into an environment variable that is evaluated when make is invoked.

We could probably do this for VERBOSE=1 as well, I did not think of doing this.

I need to have a closer look at how this affects the CMake variable QEMU_INSTANCE before approving this.

But I like this.

Reply:
Nothing is setting the CMake variable QEMU_INSTANCE today, so this looks safe.

Reply:
Works great.

Reply:
> Nothing is setting the CMake variable QEMU_INSTANCE today, so this looks safe.

So, my point is QEMU_INSTANCE was a make variable in Kbuild system, and should stay make variable in CMake build system too, there's no point of treating it as a CMake var (IMHO). There're other such variables too, I'll proceed with a similar changes then.

(I'd even argue that *any* Kbuild make variable should be whenever possible be made a make variable with CMake too, except for those for which that's not possible, like the most important ones - BAORD and CONF_FILE ).

Reply:
> Ah, so adding this '\' character will make QEMU_INSTANCE into an environment variable that is evaluated when make is invoked.

To clarify, '\' just quotes following '$', and so precludes '${FOO}' from being treated as CMake var ref, and instead gets output literally (unquoted) into the produced makefile (where make interprets its as own var reference).

Reply:
> So, my point is QEMU_INSTANCE was a make variable in Kbuild system, and should stay make variable in CMake build system too, there's no point of treating it as a CMake var (IMHO). There're other such variables too, I'll proceed with a similar changes then.

Going to have to disagree with you there. The long term goal of Zephyr is to become
more IDE friendly. Vaguely stated, if CMake is aware of a configuration, the generated
IDE can be as well, so we want to keep things in CMake when possible.

E.g. as proposed here: https://github.com/zephyrproject-rtos/zephyr/issues/4957
and here: https://github.com/zephyrproject-rtos/zephyr/issues/5140#issuecomment-346813935

I can't be more specific than that because we haven't yet figured out how this will work,
that will come after CMake has stabilized and Windows support has improved.

Reply:
> we haven't yet figured out how this will work

Sounds good. While IDE people build their shiny new world, command line people want their old stuff back ("good migration is one which is not noticed").